j2735_v2x_msgs/CMakeLists.txt

```diff
@@ -19,55 +19,97 @@ project(j2735_v2x_msgs)
 set(CMAKE_CXX_STANDARD 17)
 
 find_package(carma_cmake_common REQUIRED)
-carma_check_ros_version(2)
 carma_package()
 
-## Find ament macros and libraries
-find_package(rosidl_default_generators REQUIRED)
-find_package(ament_cmake REQUIRED)
-find_package(ament_cmake_auto REQUIRED)
-find_package(std_msgs REQUIRED)
-
-# generate messages
-file(GLOB msg_files msg/*.msg)
-set(msgs )
-foreach(filePath ${msg_files})
-  get_filename_component(name ${filePath} NAME)
-  set(msgs ${msgs} msg/${name})
-endforeach(filePath)
-
-rosidl_generate_interfaces(
-    ${PROJECT_NAME}
-        ${msgs}
+find_package(ros_environment REQUIRED)
+
+set(ROS_VERSION $ENV{ROS_VERSION})
+
+if(${ROS_VERSION} EQUAL 1) # ROS 1
+  find_package(catkin REQUIRED
+    COMPONENTS
+    message_generation
+    std_msgs
+  )
+
+  file(GLOB msg_files msg/*.msg)
+  set(msgs )
+  foreach(filePath ${msg_files})
+    get_filename_component(name ${filePath} NAME)
+    set(msgs ${msgs} ${name})
+  endforeach(filePath)
+
+  add_message_files(DIRECTORY msg
+    FILES
+    ${msgs}
+  )
+
+  generate_messages(
     DEPENDENCIES
-        std_msgs
-)
+    std_msgs
+  )
 
-install(DIRECTORY include/${PROJECT_NAME}/
-  DESTINATION include/${PROJECT_NAME}
-)
+  catkin_package(
+    CATKIN_DEPENDS
+    message_runtime
+    std_msgs
+  )
 
-ament_export_dependencies(rosidl_default_runtime)
-ament_export_include_directories(include)
-ament_package()
+  include_directories(
+  # include
+    ${catkin_INCLUDE_DIRS}
+  )
 
-install(
-  FILES rule.yml
-  DESTINATION share/${PROJECT_NAME})
+  install(DIRECTORY include/${PROJECT_NAME}/
+    DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
+    FILES_MATCHING PATTERN "*.h"
+    PATTERN ".svn" EXCLUDE
+  )
```

Comment:
NIT: These lines look awfully similar to the one we have in ROS2. We can factor it out by 

1. replacing this here with

```
set(INCLUDE_DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION})
```
2. Similarly, take out the similar `install` block from ROS2, and replace with

```
set(INCLUDE_DESTINATION include/${PROJECT_NAME})
```
3. outside the if else block, add
```

# Common for both ROS 1 and ROS 2
install(DIRECTORY include/${PROJECT_NAME}/
  DESTINATION ${INCLUDE_DESTINATION}
  FILES_MATCHING PATTERN "*.h"
  PATTERN ".svn" EXCLUDE
)

```

Reply:
Great suggestion, that looks much cleaner